### PR TITLE
Add helper function to initialize variable rate shading image

### PR DESCRIPTION
**Pull request by akioCL**
Add support for creating images that can be used in foveated rendering on XR

Additional changes on o3de [here](https://github.com/o3de/o3de/pull/13811)
ASV changes to the sample [here](https://github.com/o3de/o3de-atom-sampleviewer/pull/568)

Signed-off-by: Akio Gaule <10719597+akioCL@users.noreply.github.com>

### CODE REVIEW

File: Gems/XR/Code/Source/XRSystem.cpp [hunk 1]
```diff
@@ -370,4 +373,148 @@ namespace XR
...
+            //  _______________________________________________
+            // |____2x2____|__________2x1__________|____2x2____|
+            // |     |                                   |     |
+            // |     |                                   |     |
+            // |     |                                   |     |
+            // | 1x2 |                1x1                | 1x2 |
+            // |     |                                   |     |
+            // |_____|___________________________________|_____|
+            // |_2x2_|____2x1____|____1x1____|____2x1____|_2x2_|
+            //
```
Comment (moraaar) [inline Gems/XR/Code/Source/XRSystem.cpp:439]: Very useful comment. Awesome drawing skills 👍

**Review by moraaar — APPROVED** [review]

File: Gems/XR/Code/Source/XRSystem.cpp [hunk 2]
```diff
@@ -370,4 +373,148 @@ namespace XR
...
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 3, 0, 1, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 1, 1, 1, 2);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 0, 3, 1, 4);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 1, gridSize - 1, 2, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate1x2, 1, 3, 1, 4);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x1, 3, gridSize - 1, 2, 1);
+        }
+        break;
+        case AZ::RHI::XRFoveatedLevel::None:
+            break;
```
Comment (rgba16f) [inline Gems/XR/Code/Source/XRSystem.cpp:492]: Probably wants a comment here that says "Intentionally leave the rate grid with default values"

**Review by rgba16f — APPROVED** [review]

**Review by moudgils — APPROVED** [review]

**Review by byrcolin — APPROVED** [review]

**Comment by amzn-changml** [comment]
There are references here that can't be built without the accompanying PR in o3de merged in to development https://github.com/o3de/o3de/pull/13811 (which also fails since it defines interfaces needed in this PR). I've created a custom pipeline in AR to capture both of the associated branches in both repos here: https://jenkins.build.o3de.org/blue/organizations/jenkins/akioCL-o3de-extras/detail/vrs_image_openxr/4/pipeline/812/

This will not pass the AR gate here, however. We will need to override the AR check for this PR, merge it into development, then run the AR for the PR in the o3de repo.

@o3de/sig-core and @o3de/sig-platform: Any opinion or are we ok with doing this?

**Comment by moudgils** [comment]
I approve the override so that this PR is merged and hence clearing the path to merge the core engine PR.

**Comment by byrcolin** [comment]
I approve the override

**Comment by amzn-changml** [comment]
Overriding AR requirement. Merging this now.

**Pull request merged**